File: Lib/Telerik/Examples/RssReader/WaitingForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Telerik.WinControls.UI;
using System.Threading;

namespace RssReader
{
    public partial class WaitingForm : RadForm
    {
        private static Thread waitingThread;
        private static WaitingForm waitingForm;
        
        public WaitingForm()
        {
            InitializeComponent();
            this.radWaitingBar1.StartWaiting();
        }
 
        public static void ShowForm(Form owner) 
        {
            waitingThread = new Thread(new ParameterizedThreadStart(ThreadTask)); 
            waitingThread.IsBackground = false;
            waitingThread.Start(owner);
        } 
 
        private static void ThreadTask(object owner) 
        {
            //initialize the form
            waitingForm = new WaitingForm();
            waitingForm.ShowInTaskbar = false;
            waitingForm.Owner = (Form)owner;
            waitingForm.FormBorderStyle = FormBorderStyle.FixedDialog;
            waitingForm.ControlBox = false;
            waitingForm.TopMost = true;
            waitingForm.StartPosition = FormStartPosition.CenterScreen;            
            Application.Run(waitingForm); 
        }

        public static void CloseDialogDown()
        {
            Application.ExitThread();
        }

        public static void CloseForm() 
        {
            try
            {
                while (waitingForm == null || !waitingForm.IsHandleCreated)
                {
                    Thread.Sleep(10);
                }
                MethodInvoker mi = new MethodInvoker(CloseDialogDown);
                waitingForm.Invoke(mi);
            }
            catch { }
        }
    }
}

File: Lib/Telerik/Examples/FileExplorer/WaitingForm.vb
Imports System.ComponentModel
Imports System.Linq
Imports System.Text
Imports Telerik.WinControls.UI
Imports System.Threading

Namespace FileExplorer
	Partial Public Class WaitingForm
		Inherits RadForm
		Private Shared waitingThread As Thread
'INSTANT VB NOTE: The variable waitingForm was renamed since Visual Basic does not allow class members with the same name:
		Private Shared waitingForm_Renamed As WaitingForm

		Public Sub New()
			InitializeComponent()
			Me.radWaitingBar1.StartWaiting()
		End Sub

		Public Shared Sub ShowForm(ByVal owner As Form)
			waitingThread = New Thread(New ParameterizedThreadStart(AddressOf ThreadTask))
			waitingThread.IsBackground = False
			waitingThread.Start(owner)
		End Sub

		Private Shared Sub ThreadTask(ByVal owner As Object)
			'initialize the form
			waitingForm_Renamed = New WaitingForm()
			waitingForm_Renamed.ShowInTaskbar = False
			waitingForm_Renamed.Owner = CType(owner, Form)
            waitingForm_Renamed.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog
			waitingForm_Renamed.ControlBox = False
			waitingForm_Renamed.TopMost = True
			waitingForm_Renamed.StartPosition = FormStartPosition.CenterScreen
			Application.Run(waitingForm_Renamed)
		End Sub

		Public Shared Sub CloseDialogDown()
			Application.ExitThread()
		End Sub

		Public Shared Sub CloseForm()
            While waitingForm_Renamed Is Nothing OrElse Not waitingForm_Renamed.IsHandleCreated
                Thread.Sleep(10)
            End While
            Dim mi As New MethodInvoker(AddressOf CloseDialogDown)
            waitingForm_Renamed.Invoke(mi)
		End Sub
	End Class
End Namespace
